mempool/clist_mempool.go

```diff
@@ -446,17 +446,22 @@ func (mem *CListMempool) TxsAvailable() <-chan struct{} {
 }
 
 func (mem *CListMempool) notifyTxsAvailable() {
-	if mem.Size() == 0 {
-		panic("notified txs available but mempool is empty!")
-	}
-	if mem.txsAvailable != nil && !mem.notifiedTxsAvailable {
-		// channel cap is 1, so this will send once
-		mem.notifiedTxsAvailable = true
-		select {
-		case mem.txsAvailable <- struct{}{}:
-		default:
+	go func() {
+		if mem.Size() == 0 {
```

Comment:
I think this check should be outside of the goroutine and it should really panic.

Reply:
Oh no panic please. Who knows what happens in threading apps!